packages/headless/src/api/search/search-api-client.ts

```diff
@@ -71,6 +73,13 @@ export class SearchAPIClient {
       ...this.options,
     });
 
+    if (response instanceof Error) {
```

Comment:
Those two `if` check + the returned object are repeated very often in this PR. 

Can we have a util function for those check + what it returns ?